budgetportal/tests/test_provincial_infrastructure_projects.py

```diff
@@ -168,3 +174,141 @@ def test_other_parties(self):
         self.assertEqual(
             other_parties, ["Service Provider: DOPW\nService Provider: AAAA"]
         )
+
+
+class ProvInfraProjectAPITestCase(APITestCase):
+    def setUp(self):
+        """Create 30 Provincial Infrastructure Projects"""
+
+        self.fin_year = FinancialYear.objects.create(slug="2030-31")
+        self.url = reverse("search-provincial-infrastructure-projects")
+        self.province = u"Eastern Cape"
+        self.status = u"Construction"
+        self.source = u"Community Library Service Grant"
+        self.department = u"Department 1"
+        self.name = u"Project 1"
+        self.municipality = u"Local 2"
+        self.contractor = u"Contractor 3"
+
+        self.provinces = [
+            "Eastern Cape",
+            "Free State",
+            "Gauteng",
+            "KwaZulu-Natal",
+            "Limpopo",
+            "Mpumalanga",
+            "North West",
+            "Northern Cape",
+            "Western Cape",
+        ]
+        self.statuses = ["Design", "Tender", "Feasibility", "Construction"]
+        self.sources = [
+            "Equitable Share",
+            "Education Infrastructure Grant",
+            "Community Library Service Grant",
+        ]
+        for i in range(30):
+            ProvInfraProject.objects.create(
+                financial_year=self.fin_year,
+                IRM_project_id=i,
+                name="Project {}".format(i),
+                department="Department {}".format(i),
+                local_municipality="Local {}".format(i),
+                district_municipality="District {}".format(i),
+                province=random.choice(self.provinces),
+                status=random.choice(self.statuses),
+                primary_funding_source=random.choice(self.sources),
+                main_contractor="Contractor {}".format(i),
+                principle_agent="Principle Agent {}".format(i),
+                program_implementing_agent="Program Agent {}".format(i),
+                other_parties="Service Provider: XXXX{}".format(i),
+            )
+
+    def test_projects_per_page(self):
+        response = self.client.get(self.url, format="json")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        number_of_results = len(response.data["results"])
+        self.assertLessEqual(number_of_results, 20)
+
+    def test_filter_by_department(self):
+        project = ProvInfraProject.objects.get(department=self.department)
+
+        data = {"department": self.department}
```

Comment:
here, just hardcode the department name. Or define it in a variable inisde the test. Then it will be blatantly obvious what's happening - you're filtering for "Department 1" - and in the line above, you've also filtered for projects with "Department 1"

Reply:
variables defined for each test.

https://github.com/vulekamali/datamanager/pull/190/commits/ee17bc29b03185e791fa6fa2001619b1076a4683